Iss hipm 189

lobneroO opened the pull request:
Merge into develop

Commit:
First working version of the audio player. Audio files for tests only, remove afterwards!
Basic controlling can be: start, stop and change track - not implemented into an activity yet.

Commit:
Android Manifest update that is needed to actually start the media player as a service

Commit:
Taken out content that may have (C)

mkiehl commented on app/src/main/java/de/upb/hip/mobile/helpers/MediaPlayerService.java:
```diff
@@ -0,0 +1,112 @@
+import java.io.IOException;
+
+/**
+ * Created by Lobner on 18.04.2016.
```
Explain what class is doing

mkiehl commented on app/src/main/java/de/upb/hip/mobile/helpers/MediaPlayerService.java:
```diff
@@ -0,0 +1,112 @@
+package de.upb.hip.mobile.helpers;
```
Add license notice

mkiehl commented on app/src/main/java/de/upb/hip/mobile/helpers/MediaPlayerService.java:
```diff
@@ -0,0 +1,112 @@
+    public static final String ACTION_STOP = "de.upb.hip.mobile.STOP";
+
+
+    private int[] songList = {R.raw.audio_file_1embraceofsaturn, R.raw.audio_file_2gmanspeech,
```
Replace with Audio model class

mkiehl commented on app/src/main/java/de/upb/hip/mobile/helpers/MediaPlayerService.java:
```diff
@@ -0,0 +1,112 @@
+            mMediaPlayer.prepareAsync();
+        }
+        catch(Exception e){
+            System.out.println(e.getStackTrace());
```
Do not use System.out.println

mkiehl commented on app/src/main/java/de/upb/hip/mobile/helpers/MediaPlayerService.java:
```diff
@@ -0,0 +1,112 @@
+    }
+
+    public boolean onError(MediaPlayer mp, int what, int extra){
+        System.out.println("what: " + what + "; extra: " + extra);
```
Do not use System.out.println

mkiehl commented on app/src/main/java/de/upb/hip/mobile/helpers/MediaPlayerService.java:
```diff
@@ -0,0 +1,112 @@
+            current++;
+            current %= songList.length;
+            mMediaPlayer = MediaPlayer.create(this, songList[current]);
+        }/*catch(IOException e){
```
Exception not needed? If not, remove

mkiehl commented on app/src/main/java/de/upb/hip/mobile/helpers/MediaPlayerService.java:
```diff
@@ -0,0 +1,112 @@
+        }/*catch(IOException e){
+            System.out.println(e.getStackTrace());
+        }*/catch(Exception e){
+            System.out.println(e.getStackTrace());
```
Do not use System.out.println

mkiehl commented on app/src/main/java/de/upb/hip/mobile/helpers/MediaPlayerService.java:
```diff
@@ -0,0 +1,112 @@
+        //2. load tracks from database
+        try {
+            mMediaPlayer.reset();
+//            AssetFileDescriptor fd = getResources().openRawResourceFd(R.raw.audio_file_3gmanwise);
```
Line not needed? If not, remove?

Commit:
changed the code according to Max' annotations

Commit:
Added Audio Files for testing (The files are GEMA free. since they are my own, we have every right to use them. contact me, if you want to buy a CD ;-) )
Changed the MediaPlayer for use of Audio class
Implemented the Audio class for using preloaded media files
